### PR TITLE
Support multi-tenancy in the Java client  JobWorker API

### LINKED ISSUE #14255: The JobWorker API of the java client support multi-tenancy
### Description

The JobWorker API of the Zeebe Java client supports tenant-aware job workers by integrating the new behavior provided by #13560.

The JobWorker API should:
* Provide methods for specifying single or multiple tenant identifiers for which jobs will be polled/streamed.
* Pick up tenant IDs from the `defaultJobWorkerTenantIds` client configuration property if nothing is defined.

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Add multi-tenancy support in the Java client  JobWorker API.

## Related issues

closes #14255 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by koevskinikola** [comment]
@korthout @remcowesterhoud have a look at this PR when you have the time.

**Review by remcowesterhoud — APPROVED** [review]
LGTM! 

bors merge

**Comment by remcowesterhoud** [comment]
bors cancel

nvm we need to merge the other PR first

File: clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobWorkerBuilderStep1.java [hunk 1]
```diff
@@ -61,7 +62,8 @@ interface JobWorkerBuilderStep2 {
     JobWorkerBuilderStep3 handler(JobHandler handler);
   }
 
-  interface JobWorkerBuilderStep3 {
+  interface JobWorkerBuilderStep3 extends CommandWithOneOrMoreTenantsStep<JobWorkerBuilderStep3> {
+
```
Comment (korthout) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobWorkerBuilderStep1.java:66]: 👍 Nice catch! I completely forgot this one

**Review by korthout — APPROVED** [review]
👍 LGTM

**Comment by koevskinikola** [comment]
bors merge

**Pull request merged**